gm4_lively_lily_pads/data/gm4_lively_lily_pads/function/mechanics/cleanup/kill.mcfunction

```diff
@@ -0,0 +1,30 @@
+# cleans up block display and drops item if needed
+# @s = gm4_llp_display block_display
+# at @s
+# run from main and mechanics/attacks/candle/process_interaction
+
+# delete light
+execute if entity @s[tag=gm4_llp_light] run fill ~ ~1 ~ ~ ~1 ~ air replace light
```

Comment:
Not sure how `if block` + `setblock` compare to `fill replace`performance wise, but it shouldn't matter in this low-frequency case.